lib/vf_settokenvalue.cpp

```diff
@@ -550,14 +551,14 @@ namespace ValueFlow
                         setTokenValue(parent, std::move(result), settings);
                     } else if (Token::Match(parent, "%op%")) {
                         if (Token::Match(parent, "%comp%")) {
-                            if (!result.isFloatValue() && !value1.isIntValue() && !value2.isIntValue())
+                            if (!isFloat && !value1.isIntValue() && !value2.isIntValue())
```

Comment:
The !isFloat is redundant here.

Reply:
I thought the same but it is not (this is actually the important change based on my older comment). It reflects if *either* is a float. That doesn't mean one of them cannot be an Int.